### DISCUSSION TITLE
Importing Turtl data

### DISCUSSION THREAD

**Discussion by kjgcoop** [TriliumNext/Trilium#4056] [General]
I'm a fan of Turtl (https://github.com/turtl), but it hasn't been updated in a few years and the last OS update broke it. I exported all my notes as a big JSON file. I'd like each Turtl note to be its own Trilium note, so Trilium would have to parse the JSON. Or maybe an external script would. In any event, a-parsin' we shall go.

Anybody know of an existing way to do this?

**Comment by zadam** [comment]
Generally the easiest way is to convert it into a directory tree of HTML/Markdown files, ZIP the directory and import into Trilium.

I don't think there's any turtl specific conversion script.

**Comment by meichthys** [comment]
This should get you started: https://discourse.joplinapp.org/t/turtl-to-markdown-directory/3313

**Comment by Nriver** [comment]
Hi, I've write a python script to convert Turtl notes into markdown, https://github.com/Nriver/trilium-py/tree/main/examples/turtl-to-markdown.

Then you can import it into trilium with Trilium-py like this https://github.com/Nriver/trilium-py/tree/main#import-from-turtl.

**Comment by kjgcoop** [comment]
Thanks everybody!
